zerver/views/auth.py

```diff
@@ -320,6 +321,10 @@ def login_or_register_remote_user(request: HttpRequest, result: ExternalAuthResu
     * A zulip:// URL to send control back to the mobile or desktop apps if they
       are doing authentication using the mobile_flow_otp or desktop_flow_otp flow.
     """
+
+    for key, value in result.data_dict.get("params_to_store_in_authenticated_session", {}).items():
+        request.session[key] = value
```

Comment:
This probably deserves a comment about the fact that the `do_login` below is going to save the session? If I recall correctly, just mutating that object without an explicit refresh doesn't work...

Reply:
Hmm, I think this just works. https://docs.djangoproject.com/en/3.2/topics/http/sessions/#when-sessions-are-saved:
> Django only saves to the session database when the session has been modified – that is if any of its dictionary values have been assigned or deleted

We also use `request.session["some key"] = ...` in various places without saving e.g. `set_expirable_session_var` (and multiple other places). So this should be just fine without relying on `do_login` or anything else.

Reply:
OK, thanks for correcting me :)